File: src/rag_ingestion_factory/runtime/pilot_audit_trail.py
from __future__ import annotations


def create_audit_event(pilot_request_id: str, actor: str, from_status: str, to_status: str, reason: str) -> dict:
    return {
        "pilot_request_id": pilot_request_id,
        "actor": actor,
        "action": "review_status_changed",
        "from_status": from_status,
        "to_status": to_status,
        "reason": reason,
    }


def summarize_audit_events(events: list[dict]) -> dict[str, int]:
    return {
        "total": len(events),
        "status_changes": sum(1 for event in events if event.get("action") == "review_status_changed"),
    }

File: apps/knowledgefactory-web/lib/pilotAuditTrail.ts
export const pilotAuditTrail = {
  version: "v17.4.0",
  canonicalRule: "If a business decision changes state, the trace must survive the moment.",
  table: "public.pilot_request_audit_events",
  sampleEvents: [
    {
      pilot_request_id: "pilot-demo-001",
      actor: "AndyAI operator",
      action: "review_status_changed",
      from_status: "reviewing",
      to_status: "approved_for_pilot",
      reason: "Clear scope and measurable success criteria."
    }
  ]
};

export function summarizeAuditEvents(events = pilotAuditTrail.sampleEvents) {
  return { total: events.length, status_changes: events.filter(e => e.action === "review_status_changed").length };
}
